doc/user_guide/user_guide.md

```diff
@@ -0,0 +1,291 @@
+from exasol_data_science_utils_python.schema.column_name import ColumnName
+from exasol_data_science_utils_python.schema.column_type import ColumnType
+
+
+-- proposal write parameter into temp table in schema ...
+class CustomQueryHandler(UDFQueryHandler):
+    def __init__(self, parameter: str, query_handler_context: QueryHandlerContext):
+        super().__init__(parameter, query_handler_context)
+        self.parameter = parameter
+        self.query_handler_context = query_handler_context
+
+    def start(self) -> Union[Continue, Finish[str]]:
+        query_list = [
+          SelectQuery("SELECT 1 FROM DUAL"),
+          SelectQuery("SELECT 2 FROM DUAL")]
+        query_handler_return_query = SelectQueryWithColumnDefinition(
+            query_string="SELECT 5 AS 'return_column' FROM DUAL",
+            output_columns=[
+              Column(ColumnName("return_column"), ColumnType("INTEGER"))])
+
+        return Continue(
+            query_list=query_list,
+            input_query=query_handler_return_query)
+
+    def handle_query_result(self, query_result: QueryResult) -> Union[Continue, Finish[str]]:
+        return_value = query_result.return_column
+        result = 2 ** return_value
+        return Finish(result=f"Assertion of the final result: 32 == {result}")
+
+import builtins
+builtins.CustomQueryHandler=CustomQueryHandler # required for pickle
+
+class CustomQueryHandlerFactory:
+      def create(self, parameter: str, query_handler_context: QueryHandlerContext):
+          return builtins.CustomQueryHandler(parameter, query_handler_context)
+
+builtins.CustomQueryHandlerFactory=CustomQueryHandlerFactory
+
+from exasol_advanced_analytics_framework.udf_framework.query_handler_runner_udf \
+    import QueryHandlerRunnerUDF
+
+udf = QueryHandlerRunnerUDF(exa)
+
+def run(ctx):
+    return udf.run(ctx)
+/
+
+
+EXECUTE SCRIPT MY_SCHEMA.AAF_RUN_QUERY_HANDLER('{
+    "query_handler": {
+        "factory_class": {
+            "module": "builtins",
+            "name": "CustomQueryHandlerFactory"
+        },
+        "parameters": "bla-bla",
+        "udf": {
+            "schema": "MY_SCHEMA",
+            "name": "MY_QUERY_HANDLER_UDF"
+        }
+    },
+    "temporary_output": {
+        "bucketfs_location": {
+            "connection_name": "BFS_CON",
+            "directory": "temp"
+        },
+        "schema_name": "TEMP_SCHEMA"
+    }
+}');
+```
+
+The figure below illustrates the execution of this algorithm implemented in class `ExampleQueryHandler`.
+* When method `start()` is called, it executes two queries and an additional `input_query` to obtain the next state.
+* After the first iteration is completed, the framework calls method `handle_query_result` with the `query_result` of the `input_query` of the previous iteration.
+
+In this example, the algorithm is finished at this state and returns 2<sup>_return value_</sup> as final result.
+
+![Sample Execution](../images/sample_execution.png "Sample Execution")
```

Comment:
With the extended scenario, we need to redo the image

Reply:
We will keep the Image for now, please fix typo "Contiue". 
Image will be updated in separate PR, anyway, accordingly to the extended scenario.